modules/logging-loki-storage.adoc

```diff
@@ -0,0 +1,278 @@
+
+The recommended nomenclature for Loki storage is `logging-loki-<type>`. General secret creation format is as follows:
+//Generic secret creation
+include::snippets/logging-create-secret-snip.adoc[]
+
+You can verify a secret has been created by running the following command:
+
+[source,terminal]
+----
+oc get secrets
+----
+
+The table below displays the `type` values within the `LokiStack` CR for each storage provider. For more detailed information, consult the section for your storage provider.
+
+[options="header"]
+.Secret type quick reference
+|================================================
+| Storage provider          | Secret `type` value
+| AWS                       | s3
+| Azure                     | azure
+| Google Cloud              | gcs
+| Minio                     | s3
+| OpenShift Data Foundation | s3
+| Swift                     | swift
+| Alibaba Cloud             | alibabacloud
+|================================================
+
+== AWS S3
+
+.Prerequisites
+* Loki Operator deployed
+* Create a link:https://docs.aws.amazon.com/AmazonS3/latest/userguide/create-bucket-overview.html[bucket] on AWS.
+
+.Procedure
+* Create an Object Storage secret with keys as follows:
+
+[source,terminal]
+----
+oc create secret generic logging-loki-s3 \
+  --from-literal=bucketnames="<BUCKET_NAME>" \
+  --from-literal=endpoint="<AWS_BUCKET_ENDPOINT>" \
+  --from-literal=access_key_id="<AWS_ACCESS_KEY_ID>" \
+  --from-literal=access_key_secret="<AWS_ACCESS_KEY_SECRET>" \
+  --from-literal=region="<AWS_REGION_OF_YOUR_BUCKET>"
+----
+
+Suggested secret name: `logging-loki-s3`
+
+type: `s3`
+
+== Azure
+
+.Prerequisites
+* Loki Operator deployed
+* Create a link:https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blobs-introduction[bucket] on Azure.
+
+.Procedure
+* Create an Object Storage secret with keys as follows:
+
+[source,terminal]
+----
+oc create secret generic logging-loki-azure \
+  --from-literal=container="<AZURE_CONTAINER_NAME>" \
+  --from-literal=environment="<AZURE_ENVIRONMENT>" \
+  --from-literal=account_name="<AZURE_ACCOUNT_NAME>" \
+  --from-literal=account_key="<AZURE_ACCOUNT_KEY>"
+----
+
+Suggested secret name: `logging-loki-azure`
+
+type: `azure`
+
+
+== Google Cloud Storage
+
+.Prerequisites
+
+* Loki Operator deployed
+
+* Create a link:https://cloud.google.com/resource-manager/docs/creating-managing-projects[project] on Google Cloud Platform.
+
+* Create a link:https://cloud.google.com/storage/docs/creating-buckets[bucket] under same project.
+
+* Create a link:https://cloud.google.com/docs/authentication/getting-started#creating_a_service_account[service account] under same project for GCP authentication.
+
+.Procedure
+
```

Comment:
Please format as a multi-step procedure with line numbers (`. `) and `+` characters to align following text.